ckanext/datastore/backend/postgres.py

```diff
@@ -277,6 +277,10 @@ def _cache_types(context):
         psycopg2.extras.register_composite('nested',
                                            connection.connection,
                                            True)
+        # don't automatically convert to python objects
+        # when using native json types in 9.2+
+        # http://initd.org/psycopg/docs/extras.html#adapt-json
+        psycopg2.extras.register_default_json(loads=lambda x: x)
```

Comment:
interesting! This looks like a big performance win in many cases, but does it break anything?

Reply:
We could consider using https://github.com/ckan/ckan/blob/master/ckan/lib/lazyjson.py#L7 to wrap the string returned from postgres so that if it's pased passed to simplejson.dump it will be rendered intact.

Reply:
The change in psycopg2 (which needed to be upgraded to have pg10 support) automatically converting json to a object without a json string representation *does break* the assumptions in ckan/tests ie. about the "nested" custom type. Hence this PR to disable that behaviour.

LazyJSONObject could be considered but would need to refactor further eg. using simplejson in unit test assertions, adding a __eq__ implementation etc.

Reply:
Don't do this please, you're going to break other extensions. Either pass the cursor into `register_default_json` to scope it (which would be fine) or better yet if possible cast the JSON columns to `::text` (best performance and explicit behavior).

Reply:
Refactored to scope more closely to just datastore.

Reply:
👍